neo/io/nwbio.py

```diff
@@ -0,0 +1,810 @@
+                hierarchy = json.loads(timeseries.comments)
+            except JSONDecodeError:
+                # For NWB files created with other applications, we put everything in a single
+                # segment in a single block
+                # todo: investigate whether there is a reliable way to create multiple segments,
+                #       e.g. using Trial information
+                block_name = "default"
+                segment_name = "default"
+            else:
+                block_name = hierarchy["block"]
+                segment_name = hierarchy["segment"]
+            segment = self._get_segment(block_name, segment_name)
+            if isinstance(timeseries, AnnotationSeries):
+                event = EventProxy(timeseries, group_name)
+                if not lazy:
+                    event = event.load()
+                segment.events.append(event)
+                event.segment = segment
+            elif timeseries.rate:  # AnalogSignal
+                signal = AnalogSignalProxy(timeseries, group_name)
+                if not lazy:
+                    signal = signal.load()
+                segment.analogsignals.append(signal)
+                signal.segment = segment
+            else:  # IrregularlySampledSignal
+                signal = AnalogSignalProxy(timeseries, group_name)
+                if not lazy:
+                    signal = signal.load()
+                segment.irregularlysampledsignals.append(signal)
+                signal.segment = segment
+
+    def _read_units(self, lazy):
+        if self._file.units:
+            for id in self._file.units.id[:]:
+                try:
+                    # NWB files created by Neo store the segment and block names as extra columns
+                    segment_name = self._file.units.segment[id]
+                    block_name = self._file.units.block[id]
+                except AttributeError:
+                    # For NWB files created with other applications, we put everything in a single
+                    # segment in a single block
+                    segment_name = "default"
+                    block_name = "default"
+                segment = self._get_segment(block_name, segment_name)
+                spiketrain = SpikeTrainProxy(self._file.units, id)
+                if not lazy:
+                    spiketrain = spiketrain.load()
+                segment.spiketrains.append(spiketrain)
+                spiketrain.segment = segment
+
+    def _read_acquisition_group(self, lazy):
+        self._read_timeseries_group("acquisition", lazy)
+
+    def _read_stimulus_group(self, lazy):
+        self._read_timeseries_group("stimulus", lazy)
+
+    def write_all_blocks(self, blocks, **kwargs):
+        """
+        Write list of blocks to the file
+        """
+        # todo: allow metadata in NWBFile constructor to be taken from kwargs
+        start_time = datetime.now()
```

Comment:
I don't like setting `now` as a default for `sessions_start_time`. It's almost always wrong.

Reply:
`session_start_time` is required by NWB, so would you suggest we raise an Exception here if a value is not provided?

Reply:
I think that's probably better. I know it's not ideal, but I don't like the idea of automatically writing data which we know to be incorrect.

Reply:
`now` has been replaced.